tesseract_ros/test/kdl_env_unit.cpp

```diff
@@ -0,0 +1,118 @@
+#include <tesseract_core/macros.h>
+TESSERACT_IGNORE_WARNINGS_PUSH
+#include <ros/package.h>
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+#include <fstream>
+#include <urdf_parser/urdf_parser.h>
+TESSERACT_IGNORE_WARNINGS_POP
+
+#include "tesseract_ros/kdl/kdl_env.h"
+#include "tesseract_ros/ros_tesseract_utils.h"
+
+class TesseractKDLEnvUnit : public testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    std::string path = ros::package::getPath("tesseract_ros");
+    std::ifstream urdf_ifs(path + "/test/urdf/pppbot.urdf");
+    std::ifstream srdf_ifs(path + "/test/srdf/pppbot.srdf");
+    std::string urdf_xml_string((std::istreambuf_iterator<char>(urdf_ifs)), (std::istreambuf_iterator<char>()));
+    std::string srdf_xml_string((std::istreambuf_iterator<char>(srdf_ifs)), (std::istreambuf_iterator<char>()));
+
+    urdf_model = urdf::parseURDF(urdf_xml_string);
+    srdf_model = srdf::ModelSharedPtr(new srdf::Model);
+    srdf_model->initString(*(urdf_model.get()), srdf_xml_string);
+
+    env.reset(new tesseract::tesseract_ros::KDLEnv());
+    env->init(urdf_model, srdf_model);
+  }
+
+  bool discreteCollisionCheck()
+  {
+    tesseract::DiscreteContactManagerBasePtr dcm;
+    dcm = env->getDiscreteContactManager();
+    tesseract::ContactResultMap collisions;
```

Comment:
@Levi-Armstrong it turned out that, regarding the discrete contact manager, I was configuring the `ACM` incorrectly. The `clone` method works as expected and the `ACM` behavior is consistent. 

Still, I am getting a failure regarding the continuous collision detection, it would be awesome if you could take a look in case you I am doing anything weird in the code.
The test here is simple: `js1` is collision-free, `js2` is collision free, `js4` lies somewhere between `js1` and `js2`, and the discrete contact manager correctly finds a collision. The continuous contact manager, though, does not report any collision when going from `js1` to `js2` (a simple two-waypoints trajectory, where `traj[0,:] = js1` and `traj[1,:] = js2`

Reply:
@Levi-Armstrong apparently the continuous collision manager will work in the tests after calling `setActiveCollisionObjects`, so apparently that is not done by default on robot links by `KDLEnv`. Is there a design decision behind the behavior? How should I proceed about this? Should I provide a quick fix or just leave the continuous contact manager "disabled" by default on robot links?

Reply:
Sure I will take a loom tomorrow.

Reply:
I took a look and it should be setting them but it appears to not be. I will create a PR so that active objects are set on creation and when attached objects are added or removed.

Reply:
I believe I have fixed the issue with active links not being set in PR #64. Could you take a look and see it fixes the issue?

Reply:
Thank you working on #64. I am trying now these changes on top of the current master (which includes #64) but I am getting an assert from `bullet_cast_bvh_manager.cpp` when called by `continuousCollisionCheckTrajectory`, called in `kvl_env_unit.cpp`. Here is the stack trace:
```
#0  0x00007ffff587d428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff587f02a in __GI_abort () at abort.c:89
#2  0x00007ffff5875bd7 in __assert_fail_base (fmt=<optimized out>, 
    assertion=assertion@entry=0x7fffe75269e0 "cow->m_collisionFilterGroup == btBroadphaseProxy::KinematicFilter", 
    file=file@entry=0x7fffe75268f0 "/home/alessio/ros_ws/src/third_party/tesseract/tesseract_collision/src/bullet/bullet_cast_bvh_manager.cpp", line=line@entry=211, 
    function=function@entry=0x7fffe7528400 <tesseract::tesseract_bullet::BulletCastBVHManager::setCollisionObjectsTransform(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Eigen::Transform<double, 3, 1, 0> const&, Eigen::Transform<double, 3, 1, 0> const&)::__PRETTY_FUNCTION__> "virtual void tesseract::tesseract_bullet::BulletCastBVHManager::setCollisionObjectsTransform(const string&, const Isometry3d&, const Isometry3d&)") at assert.c:92
#3  0x00007ffff5875c82 in __GI___assert_fail (assertion=0x7fffe75269e0 "cow->m_collisionFilterGroup == btBroadphaseProxy::KinematicFilter", 
    file=0x7fffe75268f0 "/home/alessio/ros_ws/src/third_party/tesseract/tesseract_collision/src/bullet/bullet_cast_bvh_manager.cpp", line=211, 
    function=0x7fffe7528400 <tesseract::tesseract_bullet::BulletCastBVHManager::setCollisionObjectsTransform(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Eigen::Transform<double, 3, 1, 0> const&, Eigen::Transform<double, 3, 1, 0> const&)::__PRETTY_FUNCTION__> "virtual void tesseract::tesseract_bullet::BulletCastBVHManager::setCollisionObjectsTransform(const string&, const Isometry3d&, const Isometry3d&)") at assert.c:101
#4  0x00007fffe74f7d5e in tesseract::tesseract_bullet::BulletCastBVHManager::setCollisionObjectsTransform (this=0xa7ca80, name="link1", pose1=..., pose2=...)
    at /home/alessio/ros_ws/src/third_party/tesseract/tesseract_collision/src/bullet/bullet_cast_bvh_manager.cpp:211
#5  0x000000000049da45 in tesseract::continuousCollisionCheckTrajectory (manager=..., env=..., kin=..., traj=..., contacts=std::vector of length 0, capacity 1, 
    first_only=true) at /home/alessio/ros_ws/src/third_party/tesseract/tesseract_core/include/tesseract_core/basic_env.h:285
#6  0x00000000004a021c in TesseractKDLEnvUnit::continuousCollisionCheck (this=0x74d210, start=std::unordered_map with 3 elements = {...}, 
    target=std::unordered_map with 3 elements = {...}) at /home/alessio/ros_ws/src/third_party/tesseract/tesseract_ros/test/kdl_env_unit.cpp:61
#7  0x0000000000498cb9 in TesseractKDLEnvUnit_TestACMWorks_Test::TestBody (this=0x74d210)
    at /home/alessio/ros_ws/src/third_party/tesseract/tesseract_ros/test/kdl_env_unit.cpp:121
#8  0x00007ffff7bad9d8 in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void> (object=0x74d210, method=&virtual testing::Test::TestBody(), 
    location=0x7ffff7bba0c3 "the test body") at /usr/src/gtest/src/gtest.cc:2078
#9  0x00007ffff7ba8a2d in testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void> (object=0x74d210, method=&virtual testing::Test::TestBody(), 
    location=0x7ffff7bba0c3 "the test body") at /usr/src/gtest/src/gtest.cc:2114
#10 0x00007ffff7b8d7ae in testing::Test::Run (this=0x74d210) at /usr/src/gtest/src/gtest.cc:2151
#11 0x00007ffff7b8dffe in testing::TestInfo::Run (this=0x74c710) at /usr/src/gtest/src/gtest.cc:2326
#12 0x00007ffff7b8e697 in testing::TestCase::Run (this=0x74d2e0) at /usr/src/gtest/src/gtest.cc:2444
#13 0x00007ffff7b9581e in testing::internal::UnitTestImpl::RunAllTests (this=0x74cf30) at /usr/src/gtest/src/gtest.cc:4315
#14 0x00007ffff7bae931 in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (object=0x74cf30, 
    method=(bool (testing::internal::UnitTestImpl::*)(testing::internal::UnitTestImpl * const)) 0x7ffff7b9554c <testing::internal::UnitTestImpl::RunAllTests()>, 
    location=0x7ffff7bba8d0 "auxiliary test code (environments or event listeners)") at /usr/src/gtest/src/gtest.cc:2078
#15 0x00007ffff7ba98a3 in testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (object=0x74cf30, 
    method=(bool (testing::internal::UnitTestImpl::*)(testing::internal::UnitTestImpl * const)) 0x7ffff7b9554c <testing::internal::UnitTestImpl::RunAllTests()>, 
    location=0x7ffff7bba8d0 "auxiliary test code (environments or event listeners)") at /usr/src/gtest/src/gtest.cc:2114
#16 0x00007ffff7b943bc in testing::UnitTest::Run (this=0x7ffff7dd68c0 <testing::UnitTest::GetInstance()::instance>) at /usr/src/gtest/src/gtest.cc:3926
#17 0x000000000049b918 in RUN_ALL_TESTS () at /usr/include/gtest/gtest.h:2288
#18 0x000000000049b1d9 in main (argc=1, argv=0x7fffffffc4a8) at /home/alessio/ros_ws/src/third_party/tesseract/tesseract_ros/test/kdl_env_unit.cpp:181
```

Reply:
@Levi-Armstrong I found a bug in #64 (which caused the segmentation fault above), fixed it and added a test. The other tests from this PR now also pass without the need of the additional `setActiveCollisionObjects`